### PR TITLE
install and doc changes needed to introduce builtin metrics

### LINKED ISSUE #684: [Feature Request] Built-in load generator and metrics collection
**Is your feature request related to a problem? Please describe.**
Iter8 currently has a dependency on telemetry (example Prometheus) to collect basic data such as error rate and latency values for a service. If a built-in task could query the application and populate certain built in metrics for the given version(s), conformance and canary tests in Iter8 tutorials can be performed without dependence on any telemetry provider.

**Why is the feature useful for Iter8 users?**
This feature will enable users to get started with Iter8 without setting up metrics databases for telemetry.

**Describe the solution you'd like**
A task that can generate load and collect some standard built-in metrics like latency and error rates.

**Describe alternatives you've considered**
Use built in load generation without metrics collection. This nullifies significant advantages of the above proposal.

**Does this issue require a design doc/discussion? If there is a link to the design document/discussions, please provide it below.**

This task will make it possible to collect the following metrics for any version of any application.

1. Request count
2. Error count
3. Error rate
4. Mean latency
5. Median latency
6. 75th percentile tail latency
7. 90th percentile tail latency
8. 95th percentile tail latency
9. 99th percentile tail latency

For example, embedding the following task will enable collection of the above 9 metrics for default and canary versions hosted at their respective URLs. Since there is a payload URL, Iter8 will download the payload from the URL and send post requests with this payload to the two versions.

```yaml
- task: metrics/collect
   with:
    payloadURL: https://raw.githubusercontent.com/kubeflow/kfserving/master/docs/samples/v1beta1/rollout/input.json
    versions:
    - name: default
      url: http://default-version.default.svc.cluster.local
    - name: canary
      url: http://canary-version.default.svc.cluster.local
```

The task will also support a `loadOnly` option, which when set to true, will not collect any metrics but will simply generate load. Here is a variation of the above task with `loadOnly` set to true. Since there is no payload now, the request will be GET requests as opposed to POST above.

```yaml
- task: metrics/collect
   with:
    loadOnly: true
    versions:
    - name: default
      url: http://default-version.default.svc.cluster.local
    - name: canary
      url: http://canary-version.default.svc.cluster.local
```

**How will this feature be tested?**
- [x] Unit tested in handler repo
- [x] CRD changes will be unit tested in etc3 repo. In particular, etc3 interactions with analytics should not overwrite existing metrics histograms in status.
- [x] Unit tested in analytics repo
- [x] Docker image of handler will be integration tested as part of at least one tutorial which will be converted to use built-in metrics collector (eventually most tutorials will shift to builtin metrics)

**How will this feature be documented?**
- [x] iter8.tools will have this task description documented
- [x] Knative conformance tutorial & experiment will use builtin metrics

### PR DESCRIPTION
**Pull request by sriumcp**
Signed-off-by: Srinivasan Parthasarathy <spartha@us.ibm.com>

Fixes #684

### CODE REVIEW

File: mkdocs/docs/reference/tasks/metrics.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	// URL of the JSON file to send during the query; optional
+	PayloadURL *string `json:"payloadURL,omitempty" yaml:"payloadURL,omitempty"`
+	// if LoadOnly is set to true, this task will send requests without collecting metrics; optional
+	LoadOnly *bool `json:"loadOnly,omitempty" yaml:"loadOnly,omitempty"`	
+} -->
+
+| Field name | Field type | Description | Required |
+| ----- | ---- | ----------- | -------- |
+| time | string | Duration of the `metrics/collect` task run. Specified in the [Go duration string format](https://golang.org/pkg/time/#ParseDuration). Default value is `5s`. | No |
+| payloadURL | string | URL of JSON-encoded data. If this field is specified, the metrics collector will send HTTP POST requests to versions, and the POST requests will contain this JSON data as payload. | No |
```
Comment (kalantar) [inline mkdocs/docs/reference/tasks/metrics.md:68]: This says _payloadURL_. The example has _url_. 
How does this work for the bookinfo app that requires a Host header also encoded in the fortio request?

**Review by kalantar — CHANGES_REQUESTED** [review]
Looks good. Minor issue identified.

File: mkdocs/docs/reference/tasks/metrics.md (hunk 1 above)
Comment (sriumcp) [reply mkdocs/docs/reference/tasks/metrics.md:68]: @kalantar Those are two separate fields! 

URL is a sub-field within the version field. It represents the URL where  the version serves requests. It is mandatory.

payloadURL is an optional field (top level). KFServing and Seldon examples require POSTing JSON to targets. `payloadURL` enables you to specify a location for this JSON payload.

**Review by huang195 — APPROVED** [review]
/lgtm

**Pull request merged**